cpp/pipeline/PhaseAddAuthorSynonyms.cc

```diff
@@ -0,0 +1,136 @@
+#include "PhaseAddAuthorSynonyms.h"
+#include <iostream>
+#include <vector>
+#include <cstdlib>
+#include "Compiler.h"
+#include "MarcUtil.h"
+#include "MediaTypeUtil.h"
+#include "StringUtil.h"
+#include "Subfields.h"
+#include "util.h"
+
+
+static unsigned modified_count(0);
+static std::map <std::string, std::string> author_to_synonyms_map;
+static std::string field_list("100abcd:400abcd");
+static std::string primary_author_field("100abcd");
+static std::set <std::string> synonyms;
+static std::vector <std::string> tags_and_subfield_codes;
+
+
+void RemoveCommasDuplicatesAndEmptyEntries(std::vector <std::string> *const vector) {
+    std::vector <std::string> cleaned_up_vector;
+    std::set <std::string> uniqe_entries;
+
+    for (auto &entry : *vector) {
+        StringUtil::RemoveChars(",", &entry);
+
+        if (entry.empty())
+            continue;
+
+        const bool is_new_entry(uniqe_entries.emplace(entry).second);
+        if (is_new_entry)
+            cleaned_up_vector.emplace_back(std::move(entry));
+    }
+    vector->swap(cleaned_up_vector);
+}
+
+
+std::string ExtractNameFromSubfields(const std::string &field_contents, const std::string &subfield_codes) {
```

Comment:
Es abstrahiert den Vorgang einen Namen aus den Unterfeldern zu lesen. Als alleinstehende Funktion macht es so keinen Sinn, aber im Kontext in dem es genutzt wird:

    const std::string secondary_name(ExtractNameFromSubfields(fields[secondary_name_field_index], secondary_field_subfield_codes));